metrics/docker/stats_helpers.go

```diff
@@ -63,10 +64,14 @@ func collect(
 	containerName string,
 	out chan<- stats,
 ) {
-	logger.Debug("Collecting stats", "container", containerName)
+	logger.Debug("Collecting stats")
 
 	response, err := cli.ContainerStats(ctx, containerName, true)
 	if err != nil {
+		if errdefs.IsNotFound(err) {
+			logger.Warn("Container not found, metrics won't be collected for container", "error", err)
```

Comment:
Just to double check: a "not found" error is only possible when the container is not being used in the test, not because it actually didn't run (due to an error), because otherwise the test would abort?

Reply:
Right, if a container wouldn't start we wouldn't even reach this part, we wait for containers to be running and healthy when we start them.